File: Python/Regression/Salary Data_LinearRegression.py
# Import Libraries
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Import Dataset
df = pd.read_csv(r"C:\Users\snapp\Data-Projects\Datasets\Salary_Data.csv")
X = df.iloc[:,:-1].values
y = df.iloc[:,1].values

# Split data into train test part
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=1/3, random_state=0)

# Fit Simple Linear Regression 
LR = LinearRegression() 
LR.fit(X_train, y_train)

# Predict the Test set 
y_pred = LR.predict(X_test)

# Detail of the Linear Regressor 
intercept = LR.intercept_
slope = LR.coef_

# Visualising Regression - train set 
plt.scatter(X_train, y_train, c = 'red', s = 10)
plt.plot(X_train, LR.predict(X_train), c = 'green')
plt.xlabel("Experience")
plt.ylabel("Salary")
plt.title("Linear Regression Between Salary and Experience(X_train)",s=8)
plt.show()

# Visualising Regression - test set 
plt.scatter(X_test, y_test, c = 'red', s = 10)
plt.plot(X_train, LR.predict(X_train), c = 'green')
plt.xlabel("Experience")
plt.ylabel("Salary")
plt.title("Linear Regression Between Salary and Experience(X_test)",s=8)
plt.show()

File: R/Regression/SalaryData-LinearRegression.R
# Simple Linear Regression 
df = read.csv("C:/Users/snapp/Data-Projects/Datasets/Salary_Data.csv")

# Set random state for seeing same data 
set.seed(123)

# Split Dataset into train test split 
library(caTools)
split = sample.split(df$Salary, SplitRatio = 2/3)
train_set = subset(df, split == TRUE)
test_set = subset(df, split == FALSE)

# Simple Linear regression does not need to feature scaling 

# Fit Simple Linear Regression to the Training Set 
LR = lm(formula = Salary ~ YearsExperience,
        data = train_set)

print(summary(LR))

# Predict the Test Set 
y_pred = predict(LR, newdata = test_set)
print(y_pred)

# Visualise the Training set 
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = train_set$YearsExperience, y = train_set$Salary), 
             colour = 'red') + 
  geom_line(aes(x = train_set$YearsExperience, y = predict(LR, newdata = train_set)),
             colour = 'green') + 
  ggtitle('Salary vs Experience(Train set)') +
  xlab('Years of experience') + 
  ylab('Salary')

# Visualise the Training set 
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
             colour = 'red') + 
  geom_line(aes(x = train_set$YearsExperience, y = predict(LR, newdata = train_set)),
            colour = 'green') + 
  ggtitle('Salary vs Experience(Train set)') +
  xlab('Years of experience') + 
  ylab('Salary')






